cmd/boost/provider_cmd.go

```diff
@@ -97,3 +103,150 @@ var libp2pInfoCmd = &cli.Command{
 		return nil
 	},
 }
+
+var queryAskCmd = &cli.Command{
+	Name:      "query-ask",
+	Usage:     "Find a miners ask",
+	ArgsUsage: "[minerAddress]",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "peerid",
+			Usage: "specify peer ID of node to make query against",
+		},
+		&cli.Int64Flag{
+			Name:  "size",
+			Usage: "data size in bytes",
+		},
+		&cli.Int64Flag{
+			Name:  "duration",
+			Usage: "deal duration",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		afmt := NewAppFmt(cctx.App)
+		if cctx.NArg() != 1 {
+			afmt.Println("Usage: query-ask [minerAddress]")
+			return nil
+		}
+
+		maddr, err := address.NewFromString(cctx.Args().First())
+		if err != nil {
+			return err
+		}
+
+		api, closer, err := lcli.GetFullNodeAPI(cctx)
```

Comment:
Note that unlike other `boost` cmds, this one expects a Full Node, not just a Gateway.

Reply:
@dirkmc this is no longer the case, we query for asks directly over libp2p and just use the gateway to fetch multiaddrs for the miner.